File: backend/models/case.py
from typing import List, Optional
from pydantic import BaseModel, Field
from enum import Enum
from datetime import datetime

class StatusEnum(str, Enum):
    submitted = "submitted"
    processing = "processing"
    complete = "complete"

class Option(BaseModel):
    key: str
    text: str
    selected: bool

class Evidence(BaseModel):
    content: str
    page_number: int
    pdf_name: str
    event_datetime: Optional[str] = None

class Step(BaseModel):
    key: int
    question: str
    options: List[Option]
    reasoning: str
    decision: str
    next_step: str
    is_met: bool
    is_final: bool
    evidence: List[Evidence]

class Case(BaseModel):
    id: Optional[int] = None
    created_at: Optional[datetime] = Field(default_factory=datetime.now)
    status: StatusEnum = StatusEnum.submitted
    procedure_name: str
    cpt_codes: List[str]
    summary: Optional[str] = None
    is_met: bool
    is_complete: bool
    steps: List[Step]

File: frontend/types/case.ts
export enum Status {
    Submitted = "submitted",
    Processing = "processing",
    Complete = "complete"
}

export interface Option {
    key: string;
    text: string;
    selected: boolean;
}

export interface Evidence {
    content: string;
    page_number: number;
    pdf_name: string;
    event_datetime: string;
}

export interface Step {
    key: number;
    question: string;
    options: Option[];
    reasoning: string;
    decision: string;
    next_step: string;
    is_met: boolean;
    is_final: boolean;
    evidence: Evidence[];
}

export interface Case {
    case_id: string;
    created_at: Date;
    status: Status;
    procedure_name: string;
    cpt_codes: string[];
    summary: string;
    is_met: boolean;
    is_complete: boolean;
    steps: Step[];
}
